components/features/Signup/AssesseeSignup/AssesseeDetails/AssesseeDetails.tsx

```diff
@@ -0,0 +1,120 @@
+import { InputField } from "@components/shared/forms/InputField";
+import { OdiLogo } from "@components/shared/svg/OdiLogo";
+import { useAssesseeSignupStoreContext } from "@context/Signup/AssesseeSignupStoreContext";
+import styles from "./AssesseeDetails.module.css";
+import Flatpickr from "react-flatpickr";
+
+import React, { useEffect, useMemo, useRef } from "react";
+import { Button } from "@components/shared/elements/Button";
+import { SigninNotice } from "@components/shared/forms/SigninNotice";
+import { DateField } from "@components/shared/forms/DateField";
+import { emptyValidator } from "@utils/validators/emptyValidator";
+import { dateValidator } from "@utils/validators/dateValidator/dateValidator";
+import { PhoneField } from "@components/shared/forms/PhoneField";
+import { useAssesseeSignupStepContext } from "@context/Signup/AssesseeSignupStepContext";
+
+const AssesseeDetails = () => {
+  const { selectStep, selectedId } = useAssesseeSignupStepContext();
+  const { data, errors, setValue, setError } = useAssesseeSignupStoreContext();
+  const firstNameRef = useRef<HTMLInputElement>(null);
+  const lastNameRef = useRef<HTMLInputElement>(null);
+  const dateOfBirthRef = useRef<Flatpickr>(null);
+
+  const inputRefs = useMemo(
+    () => ({
+      first_name: firstNameRef,
+      last_name: lastNameRef,
+      date_of_birth: dateOfBirthRef
+    }),
+    []
+  );
+
+  useEffect(() => {
```

Comment:
Hello Alex, just as a clarification, from my understanding this effect will run when an error occurs and focuses the input element according to the error? cmiiw

Reply:
Yes shad, indeed. To be exact, it will cause the first element with error to be focused.